PokeDemo/Presenter/PokemonListPresenter.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  PokemonListPresenter.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 15/07/22.
+//
+
+import Foundation
+import UIKit
+
+protocol PokemonListPresenterDelegate: AnyObject {
+    func show(_ pokemons: [Pokemon])
+    func showAlert(title: String, message: String)
+}
+
+typealias PokemonListViewPresenterDelegate = PokemonListPresenterDelegate & UIViewController
+
+final class PokemonListPresenter {
+    
+    // MARK: Properties
+    
+    // MARK: Public
+    
+    weak var delegate: PokemonListViewPresenterDelegate?
+    
+    // MARK: Private
+    
+    private let service: ServiceRepresentable
+    
+    // MARK: Initailizers
+    
+    init(service: ServiceRepresentable = Service.shared) {
+        self.service = service
+    }
+    
+    // MARK: Exposed methods
+    
+    func viewDidLoad() {
+        getPokemons()
+    }
+    
+    func setViewDelegate(delegate: PokemonListViewPresenterDelegate) {
+        self.delegate = delegate
+    }
+    
+    func didSetupCellWith(pokemon: Pokemon, completion: @escaping (UIImage?) -> Void) {
+        service.sendRequestWithJSON(endpoint: pokemon.url!, method: .get) { response, error in
```

Comment:
if url is nil, it will crash

Reply:
Covered in 8e62e9f9cf6903a304d206bb02281454063764a0